# Análise de Área Potencial para uma nova Faculdade - Atividade 1

### Por Daniel Ruhman e Marcelo Prado, 2A


In [2]:
import pandas as pd
%matplotlib inline
import matplotlib.pyplot as plt
import matplotlib

dados = pd.read_table("pes2013A.txt",sep=" ")

In [3]:
dados

V0101  UF  V0102  V0103  V0301  V0302  V8005  V0404  V4011  V0412  \
1        2013  11     15      3      1      4     42      2    5.0      2   
2        2013  11     15      4      1      4     64      8    7.0      2   
3        2013  11     15      4      2      4     13      4    0.0      2   
4        2013  11     15      5      1      2     23      8    0.0      2   
5        2013  11     15      5      2      4     19      8    0.0      2   
6        2013  11     15      5      3      4     17      8    0.0      2   
7        2013  11     15      6      1      2     59      4    NaN      2   
8        2013  11     15      6      2      4     56      2    NaN      2   
9        2013  11     15      7      1      2     25      8    1.0      2   
10       2013  11     15      7      2      4     23      2    1.0      2   
11       2013  11     15      7      3      4      5      2    NaN      4   
12       2013  11     15      8      1      2     37      2    NaN      2   
13       2013  11     15      8      2      4     25      2    NaN      2   
14       2013  11     15      8      3      2      4      2    NaN      6   
15       2013  11     15      9      1      4     50      2    5.0      2   
16       2013  11     15      9      2      4     23      2    0.0      2   
17       2013  11     15     10      1      2     28      2    5.0      2   
18       2013  11     15     11      1      2     41      2    1.0      4   
19       2013  11     15     11      2      4     34      2    1.0      2   
20       2013  11     15     11      3      2     17      2    0.0      4   
21       2013  11     15     11      4      2     15      2    0.0      2   
22       2013  11     15     11      5      2      1      2    NaN      4   
23       2013  11     15     12      1      2     51      8    1.0      2   
24       2013  11     15     12      2      4     44      4    1.0      2   
25       2013  11     15     12      3      2      8      4    NaN      4   
26       2013  11     15     13      1      2     36      2    3.0      2   
27       2013  11     15     15      1      2     49      2    5.0      2   
28       2013  11     15     15      2      2      9      8    NaN      2   
29       2013  11     15     16      1      4     25      2    0.0      2   
30       2013  11     23      1      1      2     57      8    0.0      2   
...       ...  ..    ...    ...    ...    ...    ...    ...    ...    ...   
362526   2013  53   2148      5      5      2     10      4    0.0      4   
362527   2013  53   2148      8      1      4     33      2    NaN      2   
362528   2013  53   2148      8      2      2     43      2    NaN      4   
362529   2013  53   2148      8      3      2     12      2    0.0      4   
362530   2013  53   2148      8      4      4      9      2    NaN      4   
362531   2013  53   2148      8      5      2      7      2    NaN      4   
362532   2013  53   2148      9      1      2     42      8    0.0      4   
362533   2013  53   2148      9      2      4     38      8    0.0      4   
362534   2013  53   2148      9      3      2     18      8    0.0      4   
362535   2013  53   2148      9      4      4     15      8    0.0      4   
362536   2013  53   2148      9      5      4     14      8    0.0      2   
362537   2013  53   2148     10      1      4     18      8    NaN      2   
362538   2013  53   2148     10      2      2     23      8    NaN      4   
362539   2013  53   2148     10      3      2     50      2    NaN      4   
362540   2013  53   2148     10      4      4     39      2    NaN      4   
362541   2013  53   2148     10      5      2     21      2    0.0      4   
362542   2013  53   2148     10      6      4     22      8    0.0      2   
362543   2013  53   2148     11      1      4     35      2    0.0      2   
362544   2013  53   2148     11      2      2     49      2    5.0      4   
362545   2013  53   2148     11      3      4     13      2    0.0      4   
3625

In [7]:
dados["Regiao"] = dados.UF
dados.Regiao[dados.UF <=16]="Norte"
dados.Regiao[(dados.UF >=17) & (dados.UF <=29)]="Nordeste"
dados.Regiao[(dados.UF >=31) & (dados.UF <=35)]="Sudeste"
dados.Regiao[(dados.UF >=41) & (dados.UF <=43)]="Sul"
dados.Regiao[(dados.UF >=50) & (dados.UF <=53)]="Centro-Oeste"
dados.Regiao

/Users/daniruhman/anaconda/lib/python3.5/site-packages/ipykernel/__main__.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  from ipykernel import kernelapp as app
/Users/daniruhman/anaconda/lib/python3.5/site-packages/ipykernel/__main__.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  app.launch_new_instance()
/Users/daniruhman/anaconda/lib/python3.5/site-packages/ipykernel/__main__.py:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
/Users/daniruhman/anaconda/lib/python3.5/site-pac

1                Norte
2                Norte
3                Norte
4                Norte
5                Norte
6                Norte
7                Norte
8                Norte
9                Norte
10               Norte
11               Norte
12               Norte
13               Norte
14               Norte
15               Norte
16               Norte
17               Norte
18               Norte
19               Norte
20               Norte
21               Norte
22               Norte
23               Norte
24               Norte
25               Norte
26               Norte
27               Norte
28               Norte
29               Norte
30               Norte
              ...     
362526    Centro-Oeste
362527    Centro-Oeste
362528    Centro-Oeste
362529    Centro-Oeste
362530    Centro-Oeste
362531    Centro-Oeste
362532    Centro-Oeste
362533    Centro-Oeste
362534    Centro-Oeste
362535    Centro-Oeste
362536    Centro-Oeste
362537    Centro-Oeste
362538    C

**Filtrando os dados:**

In [ ]:
jovens = dados[(dados.V8005 >= 16) & (dados.V8005 <= 25)] #pegar apenas os jovens
brancos = jovens[jovens.V0404==2] #pegar pessoas brancas
comDiploma = brancos[brancos.V4802>=10] #pegar apenas quem tem diploma de ensino medio sem faculdade
redePrivada = comDiploma[comDiploma.V6002==4] #pegar alunos de rede privada
alfabetizados = redePrivada[redePrivada.V0601 == 1] #pegar apenas os que sabem ler

homens = alfabetizados[alfabetizados.V0302 == 2]
mulheres = alfabetizados[alfabetizados.V0302 == 4]

**Organizando os resultados**

Aqui estamos obtendo as frequências relativas de homens e mulheres em cada estado, seguidos de um gráfico ilustrativo.

In [ ]:
mulheresTotais = mulheres.UF.value_counts(True) * 100
homensTotais = homens.UF.value_counts(True) * 100

In [ ]:
mulheresTotais.plot(kind='bar',legend=False,color=('red' ,'blue'),title='Frequência relativa das mulheres em cada UF')

In [ ]:
homensTotais.plot(kind='bar',legend=False,color=('green' ,'black'),title='Frequência relativa dos homens em cada UF')

**Tabelas cruzadas**
Essa seção mostra as tabelas cruzadas criadas pelo grupo

*Cor x Idade*

In [ ]:
cor = jovens.V0404.astype('category',ordered=True)
cor.cat.categories = ("Indígena","Branca","Preta","Amarela","Parda","Não declarado")
pd.crosstab(jovens.V8005,cor)

*Cor x Rede de Ensino*

In [ ]:
rede = jovens.V6002.astype('category',ordered=True)
rede.cat.categories = ('Pública','Privada')
tabela = pd.crosstab(cor,rede)
tabela.plot.bar(stacked=True)

### Conclusões

1. A primeira conclusão desse trabalho foi perceber a importância da seleção de variáveis. Foi com certeza a parte mais trabalhosa e importante, uma vez que garante o sucesso ou o fracasso da análise, partindo do conceito *garbage in, garbage out*.


2. A segunda conclusão foi que a melhor região para Sr. Gold abrir sua universidade, segundo a análise do grupo, é a **região de São Paulo**, com cerca de *17% de homens de interesse para o negócio e 17% de mulheres*. A premissa era que a faculdade a ser aberta seria particular, e por conta disso, interesava ao Sr. Gold descobrir a região com maior concentração de jovens formados no Ensino Médio e dispostos a pagar por uma faculdade. Para isso, levou-se em consideração que esses jovens seriam brancos estudantes de escolas particules. A escolha pela cor deu-se pelo motivo do Brasil ser um país extremamente desigual, no qual a maior parte das pessoas em universidades privadas são brancas. Logo, uma futura faculdade particular tem mais chance de atrair estudantes de cor branca. 

3. A segunda região mais interessante é a do Rio Grande do Sul, com cerca de 13% de mulheres e 13% de homens de interesse ao projeto.